arch/x86/kvm/vmx/vmx.h

```diff
@@ -585,6 +602,16 @@ static inline u8 vmx_get_rvi(void)
 	 SECONDARY_EXEC_ENCLS_EXITING |					\
 	 SECONDARY_EXEC_EPT_VIOLATION_VE)
 
+#ifndef __PKVM_HYP__
+#define KVM_OPTIONAL_VMX_SECONDARY_VM_EXEC_CONTROL			\
+	(__KVM_OPTIONAL_VMX_SECONDARY_VM_EXEC_CONTROL |			\
+	 SECONDARY_EXEC_ENABLE_EPT |					\
+	 SECONDARY_EXEC_UNRESTRICTED_GUEST)
+#else /* !__PKVM_HYP__ */
+#define KVM_OPTIONAL_VMX_SECONDARY_VM_EXEC_CONTROL			\
+	__KVM_OPTIONAL_VMX_SECONDARY_VM_EXEC_CONTROL
```

Comment:
Don't we want to remove SECONDARY_EXEC_EPT_VIOLATION_VE (from both required and optional controls)?

Reply:
Or maybe it doesn't really make a difference?

Reply:
It is in optional control. we can remove it. If we don't, it should also be fine, as we didn't set up the EPT to convert a violation to #VE.